### ISSUE TITLE
Reset connection after a max age

### ISSUE THREAD

**Issue by timiblossom** [Netflix/dyno#84]
In some network environment, especially cloud, a keep-alive connection will be zombie one after some time.  It would be great to prevent a zombie connection to cripple the 99th latency.  We should have an additional background thread(s) to help out connection reset after a max keep-alive age.

**Comment by jcacciatore** [comment]
We have 2 reports from clients that after long periods of time that their app does not send any traffic, upon sending traffic all of the requests error until the connections are re-primed (triggered by the ErrorMonitor configuration). 

Digging into this I found that even though the keep alive option is set on the socket connection, the connection(s) will still become stale after the OS timeouts have expired, e.g. on Ubuntu this timeout is controlled by the following kernel parameters
```
/proc/sys/net/ipv4/tcp_keepalive_time
/proc/sys/net/ipv4/tcp_keepalive_intvl
/proc/sys/net/ipv4/tcp_keepalive_probes
/proc/sys/net/ipv4/tcp_retries2
```

There is already code to execute pings but this is enabled only for async connection pools. Rather than monitor idle connections + times and cull them I will test out enabling pings for sync connection pools.

**Comment by jcacciatore** [comment]
I have enabled pings; testing this on an instance running Ubuntu 14.04.2 LTS (trusty)

**Comment by jcacciatore** [comment]
Fix is in [PR 112](https://github.com/Netflix/dyno/pull/112)

The test mentioned in the previous comment worked, based on the following:

1. The established connections were idle (no client traffic sent) and after 20 hours netstat showed the connections were still ESTABLISHED. The linux keep alive settings are the defaults 7200, 9, 5 respectively therefore after 2 hours and 11 minutes the connections will go stale if the server isn't responding to the ACKs.

2. Started sending traffic after the 20 hour mark and there were no errors

**Comment by jcacciatore** [comment]
Merged; changes are available in release 1.4.3
